docs/ecosystem/panel.md

```diff
@@ -1,108 +1,221 @@
+    "Blue": [0, 0, 200],
+}
 
-state_options = sorted(state for state in gdf["state"].unique() if state)
+@pn.cache
+def get_data():
+    return gpd.read_file(gpd.datasets.get_path("naturalearth_cities"))
 
+gdf = get_data()
+layer = ScatterplotLayer.from_geopandas(gdf, radius_min_pixels=2, get_fill_color="red")
+cities_map = Map(layer, _height=500)
 
-def to_rgb(hex: str) -> list:
-    h = hex.strip("#")
-    return list(int(h[i : i + 2], 16) for i in (0, 2, 4))
+color_input = pn.widgets.Select(
+    name="Color", options=list(colors.keys()), description="The color of the points"
+)
 
+@pn.depends(value=color_input, watch=True)
+def set_fill_color(value):
+    cities_map.layers[0].get_fill_color = colors[value]
 
-def to_palette(cmap) -> Palette:
-    """Returns the ColorCet colormap as a palettable Palette"""
-    colors = [to_rgb(item) for item in cmap]
-    return Palette(name="colorcet", map_type="colorcet", colors=colors)
+pn.Column(
+    color_input,
+    pn.pane.IPyWidget(cities_map, width=1000),
+).servable()
+```
 
+### Build an Application
 
-def create_map(state="California", cmap=cc.fire, alpha=0.8):
-    palette = to_palette(cmap)
-    data = gdf[gdf["state"] == state]
-    layer = PathLayer.from_geopandas(data, width_min_pixels=0.8)
-    normalized_scale_rank = (data["scalerank"] - 3) / 9
-    layer.get_color = apply_continuous_cmap(normalized_scale_rank, palette, alpha=alpha)
-    map_ = Map(layer, _height=650)
-    return map_
+With Panel, you can build reusable components and styled web applications in just a few lines of code.
 
+The example below is based on the [North America Roads](../examples/north-america-roads.ipynb) example.
 
-description = """# lonboard
+![Panel lonboard application](../assets/panel-application-example.gif)
 
-A Python library for **fast, interactive geospatial vector data visualization** in Jupyter (and Panel).
+```python
+import colorcet as cc
+import geopandas as gpd
+import param
 
-By utilizing new technologies like `GeoArrow` and `GeoParquet` in conjunction with GPU-based map rendering, lonboard aims to enable visualizing large geospatial datasets interactively through a simple interface."""
+from lonboard import Map, PathLayer
+from lonboard.colormap import apply_continuous_cmap
+from lonboard._viewport import compute_view
+from palettable.palette import Palette
 
+import panel as pn
 
-# THE PANEL APP
 pn.extension("ipywidgets")
-state = pn.widgets.Select(
-    value="California",
-    options=state_options,
-    width=150,
-    name="State",
-    sizing_mode="stretch_width",
-)
-cmap = pn.widgets.ColorMap(
-    value=cc.fire,
-    options=cc.palette,
-    ncols=3,
-    swatch_width=100,
-    name="cmap by Colorcet",
-    sizing_mode="stretch_width",
-)
-alpha = pn.widgets.FloatSlider(
-    value=0.8, start=0, end=1, name="Alpha", min_width=100, sizing_mode="stretch_width"
+
+url = "https://naciscdn.org/naturalearth/10m/cultural/ne_10m_roads_north_america.zip"
+gdf = pn.state.as_cached(
```

Comment:
Is this the same as the decorator with `@pn.cache` in the previous example? Is one approach more or less common than the other? If they do the same things, it might be clearer to use the same syntax in both examples and refactor this line to a `get_data` function as well.

Reply:
I've aligned the examples to always use `@pn.cache` on a `get_data` function.